src/javascript/app/pages/cashier/cashier.js

```diff
@@ -84,43 +84,46 @@ const Cashier = (() => {
         const el_current_currency = getElementById('account_currency_current');
         const el_current_hint     = getElementById('account_currency_hint');
         const upgrade_info        = Client.getUpgradeInfo();
-        const can_change          = Client.canChangeCurrency(statement, mt5_logins);
+        const can_change          = Client.canChangeCurrency(statement, mt5_logins, dxtrade_accounts_list);
         const has_upgrade         = upgrade_info.can_upgrade || upgrade_info.can_open_multi
                                     || can_change;
         const account_action_text = has_upgrade ? `<br />${localize('[_1]Manage your accounts[_2]', [`<a href=${Url.urlFor('user/accounts')}>`, '</a>'])}` : '';
         const is_iom_client       = Client.get('residence') === 'im' || State.getResponse('website_status.clients_country') === 'im';
         const change_text_for_iom = is_iom_client ? localize('time') : localize('time or create an MT5 account');
         const url_user_account    = `<a href=${Url.urlFor('user/accounts')}>`;
 
-        const missingCriteria = (has_mt5, has_transaction) => {
+        const missingCriteria = (has_real_mt5, has_real_dxtrade, has_transaction) => {
             const existing_mt5_msg          = localize('You can no longer change the currency because you\'ve created an MT5 account.') + account_action_text;
             const existing_transaction_msg  = localize('You can no longer change the currency because you\'ve made a first-time deposit.') + account_action_text;
 
-            return has_mt5 && !has_transaction ? existing_mt5_msg : existing_transaction_msg;
+            return has_real_mt5 && has_real_dxtrade && !has_transaction ? existing_mt5_msg : existing_transaction_msg;
         };
 
         // Set messages based on currency being crypto or fiat
         // If fiat, set message based on if they're allowed to change currency or not
         // Condition is to have no MT5 accounts *and* have no transactions
-        const currency_message = Currency.isCryptocurrency(currency)
+        const currency_is_set_message   = has_no_real_mt5 && has_no_real_dxtrade && has_no_transaction
+            ? localize('Your fiat account\'s currency is currently set to [_1].', `${currency}`)
+            : localize('Your fiat account\'s currency is set to [_1].', `${currency}`);
+        const currency_message          = Currency.isCryptocurrency(currency)
             ? localize('This is your [_1] account.', `${Currency.getCurrencyDisplayCode(currency)}`)
-            : has_no_mt5 && has_no_transaction
-                ? localize('Your fiat account\'s currency is currently set to [_1].', `${currency}`)
-                : localize('Your fiat account\'s currency is set to [_1].', `${currency}`);
+            : currency_is_set_message;
 
-        const currency_hint = Currency.isCryptocurrency(currency)
+        const fiat_currency_change_hint = has_no_real_mt5 && has_no_real_dxtrade && has_no_transaction
```

Comment:
I introduced `currency_is_set_message` and `fiat_currency_change_hint` variables in order to eliminate eslint warning 'Do not nest ternary expressions'